base/sharedarray.jl

```diff
@@ -249,7 +249,7 @@ function rand!{T}(S::SharedArray{T})
 end
 
 function randn!(S::SharedArray)
-    f = S->map!(x->randn, S.loc_subarr_1d)
+    f = S->map!(x->randn(), S.loc_subarr_1d)
```

Comment:
Computing in 64 bit and converting to 32 should be mostly fine, from what I gather. But I'm ok with the proposed change anyway, it makes sense to rely on any implementation `AbstractArray` provides.